Sources/FoundationEssentials/AttributedString/AttributedStringAttributeConstrainingBehavior.swift

```diff
@@ -217,32 +240,36 @@ extension AttributedString.Guts {
             return
         }
 
-        if type == .attributes, range.isEmpty {
+        if type == .attributes, utf8Range.isEmpty {
             // For attribute-only mutations, we expand the constrained styles out from the mutated
             // range to the paragraph boundaries. If only attributes were modified and the range is
             // empty, then no true mutation occurred.
             return
         }
 
-        let strRange = utf8IndexRange(from: range)
+        let strRange = utf8IndexRange(from: utf8Range)
 
         // Character-based constraints
-        if type == .attributesAndCharacters || constraintsInvolved?._containsCharacterConstraint ?? true {
-            fixCharacterConstrainedAttributes(in: strRange)
+        if type == .attributesAndCharacters || constraintsInvolved?._containsScalarConstraint ?? true {
+            fixScalarConstrainedAttributes(in: strRange)
         }
 
         // Paragraph-based constraints
         if type == .attributes && constraintsInvolved?.contains(.paragraph) ?? true {
             // Attributes are always applied consistently, so we only need to expand outwards and not fix the range of the mutation itself
-            let paragraphStyle = _constrainedAttributes(at: range.lowerBound, with: .paragraph)
+            let paragraphStyle = _constrainedAttributes(at: utf8Range.lowerBound, with: .paragraph)
             let paragraphRange = _paragraph(in: strRange)._utf8OffsetRange
             // FIXME: It looks like this assumes that mutated attributes are consistent throughout
             // FIXME: the mutated range. This expectation should be explicitly documented.
```

Comment:
This FIXME can be resolved -- this method is only called when the mutation is affecting attributes only. Replace with a comment highlighting this.